### ISSUE TITLE
Fix Request: false positive in NonComparableSetKeys for Java

### ISSUE THREAD

**Issue by stokpop** [jborgers/PMD-jPinpoint-rules#618]
```
public class Foo {

    public static final Comparator<Class> CLASS_NAME_COMPARATOR = Comparator.comparing(Class::getName);

    public void bar() {
        final Set<Class> types = new TreeSet<>(CLASS_NAME_COMPARATOR);
        foo(types);
    }

    public void foo(Set<Class> types) { // <--- false positive, has constructor with comparator
        types.contains(String.class);
    }

}
```

Should we check all type declarations? In case above we can check the TreeSet construction in same class, but that is not always the case.

The constructor itself is also flagged, so maybe not flag declarations? 

(Same for NonComparableMapKeys rule?)
